tests/templates/test_integration.py

```diff
@@ -176,558 +138,347 @@
+    def test_cv_qnode_primary_args(self, template, diffable, nondiffable,
+                                   interface, to_var, gaussian_device_2_wires):
         """Tests integration of cv templates passing differentiable arguments as positional arguments to qnode."""
 
         # Extract keys and items
-        keys_diffable1 = [*diffable1]
-        diffable1 = list(diffable1.values())
-        keys_diffable2 = [*diffable2]
-        diffable2 = list(diffable2.values())
-
-        # Combine diffable inputs to allow passing with *
-        diffable = diffable1 + diffable2
+        keys_diffable = [*diffable]
+        diffable = list(diffable.values())
 
         # Turn into correct format
         diffable = [to_var(i) for i in diffable]
 
         # Generate qnode
-        circuit = qnode_cv_args(gaussian_device_2_wires, interface, template1, template2, len(diffable1))
+        @qml.qnode(gaussian_device_2_wires, interface=interface)
+        def circuit(*diffable, keys_diffable=None, nondiffable=None):
+            # Turn diffables back into dictionary
+            dict = {key: item for key, item in zip(keys_diffable, diffable)}
+
+            # Merge with nondiffables
+            dict.update(nondiffable)
+
+            # Add number of wires
+            dict['wires'] = range(2)
+
+            template(**dict)
+            return qml.expval(qml.Identity(0))
 
         # Check that execution does not throw error
-        circuit(*diffable, keys_diffable1=keys_diffable1, keys_diffable2=keys_diffable2,
-                nondiffable1=nondiffable1, nondiffable2=nondiffable2)
+        circuit(*diffable, keys_diffable=keys_diffable, nondiffable=nondiffable)
 
-    @pytest.mark.parametrize("template1, diffable1, nondiffable1", QUBIT_DIFFABLE_NONDIFFABLE)
-    @pytest.mark.parametrize("template2, diffable2, nondiffable2", QUBIT_DIFFABLE_NONDIFFABLE)
+    @pytest.mark.parametrize("template, diffable, nondiffable", QUBIT_DIFFABLE_NONDIFFABLE)
     @pytest.mark.parametrize("interface, to_var", INTERFACES)
-    def test_integration_qubit_nondiffable(self, template1, diffable1, nondiffable1,
-                                           template2, diffable2, nondiffable2,
-                                           interface, to_var):
-        """Tests integration of qubit templates passing differentiable arguments as keyword arguments to qnode."""
+    def test_qubit_qnode_auxiliary_args(self, template, diffable, nondiffable, interface, to_var):
+        """Tests integration of qubit templates passing differentiable arguments as auxiliary arguments to qnode."""
 
         # Change type of differentiable arguments
-        # TODO: templates should all take arrays AND lists, at the moment this is not the case
-        diffable1 = {k: np.array(v) for k, v in diffable1.items()}
-        diffable2 = {k: np.array(v) for k, v in diffable2.items()}
+        # Fix: templates should all take arrays AND lists, at the moment this is not the case
```

Comment:
Is it the case that some take only arrays and some take only lists?

Reply:
Yes, I think BasisStatePrep required an array...we should investigate this systematically, allow for both and add a test to each template which uses lists or arrays for the diffable arguments!